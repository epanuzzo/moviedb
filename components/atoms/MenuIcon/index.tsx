import React from "react";

type MenuIconProps = {
  toggleMenu: () => void;
};

/**
 * MenuIcon component renders a button with a hamburger menu icon.
 * It is typically used to toggle the visibility of a menu in a responsive design.
 *
 * @component
 * @param {MenuIconProps} props - The props for the MenuIcon component.
 * @param {function} props.toggleMenu - The function to call when the button is clicked.
 * @returns {JSX.Element} The rendered MenuIcon component.
 */
const MenuIcon: React.FC<MenuIconProps> = ({ toggleMenu }) => {
  return (
    <button className="ml-2 md:hidden text-white" onClick={toggleMenu}>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 20 20"
        fill="currentColor"
        className="w-6 h-6"
      >
        <path
          fillRule="evenodd"
          d="M3 4h14a1 1 0 0 1 0 2H3a1 1 0 0 1 0-2zm0 5h14a1 1 0 0 1 0 2H3a1 1 0 0 1 0-2zm0 5h14a1 1 0 0 1 0 2H3a1 1 0 0 1 0-2z"
        />
      </svg>
    </button>
  );
};

export default MenuIcon;
