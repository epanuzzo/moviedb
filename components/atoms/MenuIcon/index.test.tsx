import React from "react";
import renderer from "react-test-renderer";
import MenuIcon from "./index";

describe("MenuIcon", () => {
  it("renders component", () => {
    const tree = renderer.create(<MenuIcon toggleMenu={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
