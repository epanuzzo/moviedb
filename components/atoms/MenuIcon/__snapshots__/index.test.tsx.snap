// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuIcon renders component 1`] = `
<button
  className="ml-2 md:hidden text-white"
  onClick={[MockFunction]}
>
  <svg
    className="w-6 h-6"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 4h14a1 1 0 0 1 0 2H3a1 1 0 0 1 0-2zm0 5h14a1 1 0 0 1 0 2H3a1 1 0 0 1 0-2zm0 5h14a1 1 0 0 1 0 2H3a1 1 0 0 1 0-2z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
